app/src/main/java/pl/futurecollars/invoicing/controller/InvoiceController.java

```diff
@@ -0,0 +1,54 @@
+package pl.futurecollars.invoicing.controller;
+
+import java.util.List;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import pl.futurecollars.invoicing.db.memory.FileBasedDatabase;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.service.InvoiceService;
+
+@RestController
+@RequestMapping("invoices")
+public class InvoiceController {
+
+    private final InvoiceService invoiceService = new InvoiceService(new FileBasedDatabase());
+
+    @PostMapping
+    public int add(@RequestBody Invoice invoice) {
+        return invoiceService.save(invoice);
+    }
+
+    @GetMapping
+    public List<Invoice> getAll() {
+        return invoiceService.getAll();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Invoice> getById(@PathVariable int id) {
+        return invoiceService.getById(id)
+                .map(invoice -> ResponseEntity.ok().body(invoice))
```

Comment:
It is ok, but in general we don;t want the logic in controllers, you could create some kind of "responseGenerator" or converter or smth like this and from that return exact type required by controller return type